### PR TITLE
Add Vapor 5 post

### PR DESCRIPTION
**Pull request by 0xTim**
Add a post about Vapor 5

### CODE REVIEW

File: Content/posts/the-future-of-vapor.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+Work has [now begun](https://github.com/vapor/vapor/pull/3229) on the next version of Vapor, Vapor 5. Whilst the APIs are very much up for change, we have some high-level goals we want to hit. Our absolute goal is for Vapor to not only be one of the de facto frameworks when building Swift applications, but a de facto choice for building a backend in _any_ language.
+
+The main goals include:
+
+* The same, great, Vapor experience
+* Full structured concurrency
+* Full up-to-date ecosystem integration
+* Provide a foundation for a modern backend
+* Rewrite of the Websocket and the MutliparkKit APIs
```
Comment (fpseverino) [inline Content/posts/the-future-of-vapor.md:29]:
```suggestion
* Rewrite of the WebSocket and the MultipartKit APIs
```

**Review by gwynne — APPROVED** [review]

File: Content/posts/the-future-of-vapor.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+* Full structured concurrency
+* Full up-to-date ecosystem integration
+* Provide a foundation for a modern backend
+* Rewrite of the WebSocket and the MultipartKit APIs
+
+### The Same Vapor Experience
+
+Vapor has always been known for providing a great developer experience with an API that's easy to work with and _feels_ Swifty. We want to ensure this stays the same and make it even better. This includes APIs that make sense, documentation that is clear and tutorials that are easy to follow, based on our fantastic community.
+
+This is what has made Vapor so popular and that is not going to change. We want to try and build on Swift's ethos of progressive disclosure, so whilst it's easy to get started with Vapor and simple to write a basic API, the hooks to write more complicated and complex applications are there are you're not constrained by the framework.
```
Comment (fpseverino) [inline Content/posts/the-future-of-vapor.md:35]:
```suggestion
This is what has made Vapor so popular and that is not going to change. We want to try and build on Swift's ethos of progressive disclosure, so whilst it's easy to get started with Vapor and simple to write a basic API, the hooks to write more complicated and complex applications are there and you're not constrained by the framework.
```

Maybe? Not sure

Comment (0xTim) [reply Content/posts/the-future-of-vapor.md:35]: Yep!

File: Content/posts/the-future-of-vapor.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+### Full Up-to-Date Ecosystem Integration
+
+As well as Swift service lifecycle, there are a number of new libraries we can take advantage of. There's a new HTTP Server we can build on top of, based on Adam's great work from Hummingbird. There's a new Middleware package being worked on that will make it easy for us to share middlewares between different frameworks.
+
+There's also the [new HTTP Types library](https://github.com/apple/swift-http-types) and Swift Argument Parser we can build on. Less stuff for Vapor to maintain and better integration with the rest of the Swift ecosystem - a win-win.
+
+### Provide a Foundation for a Modern Backend
+
+Vapor is used by companies big and small powers some very large applications. We want to ensure that Vapor not only works well for small, simple APIs but also the most demanding backends. This includes ensuring we have a full observability journey, with logging, metrics and tracing all supported out of the box.
```
Comment (fpseverino) [inline Content/posts/the-future-of-vapor.md:53]:
```suggestion
Vapor is used by companies big and small and powers some very large applications. We want to ensure that Vapor not only works well for small, simple APIs but also the most demanding backends. This includes ensuring we have a full observability journey, with logging, metrics and tracing all supported out of the box.
```

File: Content/posts/the-future-of-vapor.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+The new HTTP server will allow us to easily provide first class support for gRPC, async body streaming and SSE. OpenAPI will also be a first class system, both from generating documentation from routes and generating routes from an OpenAPI spec.
+
+And of course, with structured concurrency and Sendable, threading issues and data races are a thing of the past. Vapor 5 will provide a highly performant foundation to satisfy the most demanding users.
+
+### Rewrite of the Websocket and MultiparkKit APIs
```
Comment (fpseverino) [inline Content/posts/the-future-of-vapor.md:59]:
```suggestion
### Rewrite of the WebSocket and MultipartKit APIs
```

File: Content/posts/the-future-of-vapor.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+The journey to Vapor 5 will follow a rough plan:
+
+1. Remove all the old future-based APIs (this is already done)
+2. Implement the 'new' service architecture, AKA dependency injection
+3. Fix all the tech debt with Sendable, migrate to value type Request and Responses etc
+3. Switch out the HTTP server to provide a full async stack
+4. Migrate to Swift Service Lifecycle
+5. Migrate to Swift Testing
+6. Rewrite the Websocket and MultiparkKit APIs
```
Comment (fpseverino) [inline Content/posts/the-future-of-vapor.md:81]:
```suggestion
6. Rewrite the WebSocket and MultipartKit APIs
```

**Pull request merged**